### ISSUE TITLE
Redundancy / Exhaustiveness doesn't work with parenthesized types

### ISSUE THREAD

**Issue by Negabinary** [hazelgrove/hazel#1380]
![image](https://github.com/user-attachments/assets/0d45d077-949c-461a-b2f9-50cf3f4e1f6d)

```
type T = (+A+B) in
let x : T =  in
case x
  | A => 1
  | B => 2 
end
```

**Comment by cyrus-** [comment]
@pigumar1 do you have capacity to take a look?

**Comment by Negabinary** [comment]
@cyrus I think I alrealdy fixed this in #1382
